### ISSUE TITLE
Add form validations to all create flows

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#841]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

Something that's needed is for all create flows to have form validations such that submission is not possible without certain fields being filled. The following flows need validation:

- `organizations/create`
- `groups/create`
- 'resources/create'

**Comment by amilliiee** [comment]
I've been doing some digging during work today and I might be able to get this done if you haven't resolved this in any way yet.

**Comment by andrewtavis** [comment]
Would be happy to get the help, @amilliiee! Will assign you now, and let me know if there's anything I can do to help 😊

**Comment by amilliiee** [comment]
I think right now the only question I have is how we want this organized. I'm assuming a single validation file?

**Comment by andrewtavis** [comment]
Would be great, @amilliiee 😊 Thanks for checking in!

**Comment by andrewtavis** [comment]
Hey @amilliiee 👋 Looping @to-sta into this discussion as he's working on the login for the backend :) Let us know if you're definitely going with formkit and what the plans are 😊

**Comment by amilliiee** [comment]
@andrewtavis @to-sta I just figured since it can handle validation on it's own, it would cut down on what we'd need to do to get it working :)

I've been playing around at work and [this](https://codesandbox.io/p/devbox/sad-star-jk3qp8) devbox is a messy first attempt. Since I don't have everything I can't do a full mockup of the create groups page but the general idea is there. There's also a way to condense all of the styling but I need to look into it more.

The topic selection requires the pro version, so if we really do want to go this route I have no problem making the one time payment for it.

Let me know what you guys think, and I can have another go at validation without formkit when I get home today!

**Comment by andrewtavis** [comment]
Hey @amilliiee 👋 Sorry for not checking this sooner :) Was gone for a week, and lots to do at work once I was back. Will check this over the weekend, and also feel free to drop into the dev sync tomorrow where we can discuss this as a group 😊

CC @momanyisamuel, this is the issue that's looking at form validation :)

**Comment by andrewtavis** [comment]
Hey @amilliiee 👋 We're trying to check the sandbox within the dev sync now, but then we're getting something went wrong... Would you be able to open a PR with the changes? :)

**Comment by andrewtavis** [comment]
Sorry for the wait here :) Very busy week on my end. [VeeValidate](https://vee-validate.logaretm.com/v4/) is still the running decision for this. Would be great if you picked it up, @cquinn540!

**Comment by andrewtavis** [comment]
Assigning @nicki182 as in discussing the work for #1291 we're realizing that the form components for that need to be standardized. First work here is a rework of all of the form components such that they're ready to be used with VeeValidate (including as much HeadlessUI usage as possible/reasonable), and then from there we'll move to form validation and then back to #1291 :)

**Comment by andrewtavis** [comment]
To finalize here, the goal would be that we have all forms that are currently active using the new form components and being fully validated. The forms in question would be sign in, sign up and organization creation. Other forms likely need to be reworked, so let's not worry about them now :) Agreement from a call that @nicki182 and I are on right now: We'll do one PR that's for the components and a template form component and then from there we'll do another PR to finalize the three form pages mentioned here 😊

**Comment by andrewtavis** [comment]
Adding some notes for what I'm seeing should still be done to close this once #1338 is merged:

- [x] The password match for sign up isn't working
- [x] The sign in e2e tests are not working, but that's because we've switched over to a universal form component
    - They should be reworked to accept these new components 

CC @nicki182 :)

**Comment by andrewtavis** [comment]
Closed by #1335, #1338 and #1343 🚀🚀🚀 Thanks so much for the help here, @nicki182! 😊

**Comment by andrewtavis** [comment]
Note that we'll be making more issues now for individual forms. As of now we're done with sign in, sign up and organization creation. The first such issue is #1342 :)
